src/main/java/gg/solarmc/kitpvp/kill/DamageHolder.java

```diff
@@ -66,4 +65,8 @@ public void damage(Player damager) {
         return pepee;
     }
 
+    @Override
+    public void removeHolder(UUID uuid) {
+        damagers.remove(uuid); //don't cancel the task simply remove the link and let java gc clean it up whenever it finishes
```

Comment:
Running tasks will not be garbage collected, because the BukkitScheduler will hold a reference to them. You will indeed need to cancel any removed tasks.

Reply:
The bukkit scheduler will cancel it on completion and let it be gc'd, otherwise the bukkitscheduler itself would be a memory leak - e.g. not my problem!

Reply:
Good then